juju/juju
# Reacting to CAAS Operator changes.

## Description of change

React and report on operator pod changes, funnel status changes from the operator back into theJuju data model.

This will allow Juju to surface error message encountered when spinning up an operator for a CAAS charm deploy.

## QA steps

Manually verify the status data is stored in the DB using the mongo console.
(Modified steps taken from 
  - Boostrap and setup a caas model as per [this post on discourse](https://discourse.jujucharms.com/t/storage-support/154).
  - BEFORE deploying the charm, poison the operator image config so that the operator pod will error (needs to be a valid docker url):
    - For instance I use: ```juju controller-config caas-operator-image-path=veebers/caap-operator@sha256:29498cb4c1f89e28b3f45df6c222e3b404586c56f1fe1c79978398e89b3e0309```
  - Use this script ([found on the wiki here]()) to gain access to the mongo console and check the status data is present:
    - ```db.statuses.find().pretty()``` 
    - We're looking for an entry with an ID like: ```<model uuid>:a#mariadb#operator``` and a message like: ```Error: ImagePullBackOff```

If you repeat the above but dont posion the operator-image you will see the status as 'active'.

## Documentation changes

None.

## Bug reference

N/A
